domain/access/state/types.go

```diff
@@ -54,17 +55,28 @@ type dbUser struct {
 }
 
 // toCoreUser converts the state user to a core user.
-func (u dbUser) toCoreUser() coreuser.User {
+func (u dbUser) toCoreUser() (coreuser.User, error) {
+	name, err := coreuser.NewName(u.Name)
+	if err != nil {
+		return coreuser.User{}, errors.Annotate(err, "user name from db")
+	}
+	var creatorName coreuser.Name
+	if u.CreatorName != "" {
+		creatorName, err = coreuser.NewName(u.CreatorName)
+		if err != nil {
+			return coreuser.User{}, errors.Annotate(err, "creator name from db")
+		}
+	}
```

Comment:
If this does error out here, I wonder what the path forward to fixing it would be. I guess direct DB access to rectify it.

Reply:
Yea, as @tlm has said before, we should have the guarantee that the stuff in the DB is valid and good when we add it. I would kindof like not to have to do validation at all here, but that would mean having a seperate method to create a username without checking if its valid, and I don't really want to do that. This seems like an ok compromise